tools/ports/__init__.py

```diff
@@ -427,9 +428,13 @@ def dfs(node):
   return stack
 
 
-def resolve_dependencies(port_set, settings):
+def resolve_dependencies(port_set, settings, cflags_only=False):
   def add_deps(node):
-    node.process_dependencies(settings)
+    sig = signature(node.process_dependencies)
+    if len(sig.parameters) == 2:
+      node.process_dependencies(settings, cflags_only)
+    else:
+      node.process_dependencies(settings)
```

Comment:
Maybe add some comment here? I don't have an initial guess as to what "cflags-only" means in this context. Also, what is magical/special about having exactly two parameters..?